test/python/compiler/test_transpiler.py

```diff
@@ -797,14 +797,84 @@ def test_transpiled_basis_gates_calibrations(self):
         circ.add_calibration("h", [0], q0_x180)
 
         backend = FakeAlmaden()
-        # TODO: Remove L803-L804 in the next PR
         transpiled_circuit = transpile(
             circ,
             backend=backend,
-            basis_gates=backend.configuration().basis_gates
-            + list(circ.calibrations.keys()),
         )
         self.assertEqual(transpiled_circuit.calibrations, circ.calibrations)
+        # self.assertEqual(transpiled_circuit, circ)
+
+    def test_transpile_calibrated_custom_gate_on_diff_qubit(self):
+        """Test if the custom, non calibrated gate raises QiskitError."""
+        custom_180 = Gate("mycustom", 1, [3.14])
+
+        circ = QuantumCircuit(2)
+        circ.append(custom_180, [0])
+
+        with pulse.build() as q0_x180:
+            pulse.play(pulse.library.Gaussian(20, 1.0, 3.0), pulse.DriveChannel(0))
+
+        # Add calibration
+        circ.add_calibration(custom_180, [1], q0_x180)
+
+        backend = FakeAlmaden()
+        with self.assertRaises(QiskitError):
+            transpile(circ, backend=backend)
+
+    def test_transpile_calibrated_nonbasis_gate_on_diff_qubit(self):
+        """Test if the non-basis gates are transpiled if they are on different qubit that
+        is not calibrated."""
+        circ = QuantumCircuit(2)
+        circ.h(0)
+        circ.h(1)
+
+        with pulse.build() as q0_x180:
+            pulse.play(pulse.library.Gaussian(20, 1.0, 3.0), pulse.DriveChannel(0))
+
+        # Add calibration
+        circ.add_calibration("h", [1], q0_x180)
+
+        backend = FakeAlmaden()
+        transpiled_circuit = transpile(
+            circ,
+            backend=backend,
+        )
+        self.assertEqual(transpiled_circuit.calibrations, circ.calibrations)
+        self.assertEqual(set(transpiled_circuit.count_ops().keys()), {'u2', 'h'})
+
+    def test_transpile_subset_of_calibrated_gates(self):
+        """Test transpiling a circuit with both basis gate (not-calibrated) and
+        a calibrated gate."""
+        x_180 = Gate('mycustom', 1, [3.14])
+
+        circ = QuantumCircuit(2)
+        circ.h(0)
+        circ.append(x_180, [0])
+
+        with pulse.build() as q0_x180:
+            pulse.play(pulse.library.Gaussian(20, 1.0, 3.0), pulse.DriveChannel(0))
+
+        circ.add_calibration(x_180, [0], q0_x180)
+        circ.add_calibration('h', [1], q0_x180)  # 'h' is calibrated on qubit 1
+
+        transpiled_circ = transpile(circ, FakeAlmaden())
+        self.assertEqual(set(transpiled_circ.count_ops().keys()), {'u2', 'mycustom'})
+
+    def test_transpile_gate_calibrated_on_a_diff_qubit(self):
+        """Test transpiling a circuit with gate calibrated on a different qubits."""
```

Comment:
according to this description, this should have this line
```
        circ.add_calibration('h', [1], q0_x180)  # 'h' is calibrated on qubit 1
```
(which would then make it the same as the last test). Maybe replace this with a test where `x180` custom gate is calibrated on one qubit but not another to assert that an error is raised even when the custom operation is defined on some gates?

Reply:
Fixed in https://github.com/Qiskit/qiskit-terra/pull/5080/commits/0ce6ad2f700d92fb0a6eed324fc7506e5e988b71